### PR TITLE
feat: add provisional create canister with cycles method

### PR DESCRIPTION
**Pull request by nathanosdev**
# Motivation

This adds an additional method that I missed in the previous PR. This one is actually the most important for my use case in automation testing :).

# Changes

The provisional add canister with cycles method has been added the management canister interface.

# Tests

Unit tests have been added.

### CODE REVIEW

File: packages/ic-management/src/ic-management.canister.ts [hunk 1]
```diff
@@ -175,4 +176,28 @@ export class ICManagementCanister {
...
     this.service.delete_canister({ canister_id: canisterId });
+
+  /**
+   * Creates a canister. Only available on development instances.
+   *
+   * @param {Object} params
+   * @param {Principal} params.canisterId
+   * @param {BigInt} params.amount
+   * @param {CanisterSettings} params.settings
+   * @returns
```
Comment (peterpeterparker) [inline packages/ic-management/src/ic-management.canister.ts:187]: Doesn't return a principal?

File: packages/ic-management/src/ic-management.canister.ts [hunk 2]
```diff
@@ -175,4 +176,28 @@ export class ICManagementCanister {
    */
   deleteCanister = (canisterId: Principal): Promise<void> =>
     this.service.delete_canister({ canister_id: canisterId });
+
+  /**
+   * Creates a canister. Only available on development instances.
```
Comment (peterpeterparker) [inline packages/ic-management/src/ic-management.canister.ts:181]: dumb question: what's "development instances"? testnet?

File: packages/ic-management/src/ic-management.canister.spec.ts [hunk 3]
```diff
@@ -361,4 +361,50 @@ describe("ICManagementCanister", () => {
...
+
+      await icManagement.provisionalCreateCanisterWithCycles();
+
+      expect(
+        service.provisional_create_canister_with_cycles
+      ).toHaveBeenCalledWith({
+        amount: [],
+        settings: [
+          {
+            controllers: [],
```
Comment (peterpeterparker) [inline packages/ic-management/src/ic-management.canister.spec.ts:388]: not a must but, should we maybe also add a test with some parameters value? as we use `toNullable` to map those, maybe would be worth it? just an idea

**Review by peterpeterparker — COMMENTED** [review]
Looks great. Few ultra minor comments and we can merge it.

File: packages/ic-management/src/ic-management.canister.ts (hunk 1 above)
Comment (nathanosdev) [reply packages/ic-management/src/ic-management.canister.ts]: Ah thanks I missed that! Updated now

File: packages/ic-management/src/ic-management.canister.ts (hunk 2 above)
Comment (nathanosdev) [reply packages/ic-management/src/ic-management.canister.ts:181]: I'm not sure about testnets to be honest, but the DFX replica is definitely a development instance. That's where I'm using this.

File: packages/ic-management/src/ic-management.canister.spec.ts (hunk 3 above)
Comment (nathanosdev) [reply packages/ic-management/src/ic-management.canister.spec.ts]: Yes that's a great point, I've updated the test now, thanks!

File: packages/ic-management/src/ic-management.canister.ts (hunk 2 above)
Comment (peterpeterparker) [reply packages/ic-management/src/ic-management.canister.ts:181]: Ah in that sense, gotcha!

**Review by peterpeterparker — APPROVED** [review]
LGTM, thanks!

Side note: if you've got a cool idea about how we can communicate the developer (with an assertion?)  that `provisionalCreateCanisterWithCycles` is only available locally, I'm not against.

I don't have an idea right now but that can be a nice improvement, that way we should not assume the dev reads the doc 😉.

**Pull request merged**